File: 2965. Find Missing and Repeated Values/Find Missing and Repeated Values.py
grid = [[1,3],[2,2]]
n = len(grid)
n*=n
visit = [0]*n
s = 0
sol = [0,0]
for i in grid:
    for j in i:
        if visit[j-1]:
            sol[0] = j
            s-=j
        visit[j-1] = 1
        s+=j
sol[1] = (n*(n+1))/2 - s
print(sol)

File: 2965. Find Missing and Repeated Values/FindMissingAndRepeatedValues.java
public class FindMissingAndRepeatedValues {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length, sum = 0;
        n*=n;
        int[] visited = new int[n], sol = new int[2];
        for (int[] row:grid){
            for(int nums:row){
                if(visited[nums-1] == 1){
                    sol[0] = nums;
                    sum-=nums;
                }
                visited[nums-1] = 1;
                sum+=nums;
            }
        }
        sol[1] = n*(n+1)/2 - sum;
        return sol;
    }
    public static void main(String[] args) {
        FindMissingAndRepeatedValues fValues = new FindMissingAndRepeatedValues();
        System.out.println(fValues.findMissingAndRepeatedValues(new int[][]{{1, 3},{2, 2}}));
    }
}
